docs/cancellation.rst

```diff
@@ -153,3 +153,19 @@ The problem with this code is that it violates structural concurrency: what happ
 task raises an exception? The host task would be cancelled as a result, but the host task might be
 long gone by the time that happens. Even if it weren't, any enclosing ``try...except`` in the
 generator would not be triggered. In other words: it's a Really Bad Idea to do this!
+
+Async generators decorated with ``@asynccontextmanager`` to serve as the template for an async context
```

Comment:
lifted from trio https://trio.readthedocs.io/en/stable/reference-core.html#cancel-scopes-and-nurseries